soc/arm/silabs_sim3/sim3u/soc.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023 GARDENA GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/irq.h>
+
+#include <SI32_CLKCTRL_A_Type.h>
+#include <SI32_PBCFG_A_Type.h>
+#include <SI32_PBSTD_A_Type.h>
+#include <SI32_RSTSRC_A_Type.h>
+#include <SI32_VMON_A_Type.h>
+#include <SI32_WDTIMER_A_Type.h>
+#include <si32_device.h>
+#include <silabs_crossbar_config.h>
+
+static void gpio_init(void)
+{
+	/* Enable clocks that we need in any case */
+	SI32_CLKCTRL_A_enable_apb_to_modules_0(
+		SI32_CLKCTRL_0, SI32_CLKCTRL_A_APBCLKG0_PB0 | SI32_CLKCTRL_A_APBCLKG0_FLASHCTRL0);
+
+	/* Enable misc clocks.
+	 * We may or may nor need all of that but it includes some basics like
+	 * LOCK0, WDTIMER0 and DMA. Doing it here prevents the actual drivers
+	 * needing to know whoelse needs one of these clocks.
+	 */
+	SI32_CLKCTRL_A_enable_apb_to_modules_1(
+		SI32_CLKCTRL_0, SI32_CLKCTRL_A_APBCLKG1_MISC0 | SI32_CLKCTRL_A_APBCLKG1_MISC1);
+	SI32_CLKCTRL_A_enable_ahb_to_dma_controller(SI32_CLKCTRL_0);
```

Comment:
Could we read which clocks we need from the devicetree?

Reply:
How exactly would that look like? The proper way would probably be to have e driver+device for every clock and then pass it via devicetree and create a custom driver interface to turn the clock handle into a register for apb. 

To me that seems like a lot of additional code and overhead for something that we currently don't support changing anyway.